challengeMk2/DataBase/SQLiteDataService.cs

```diff
@@ -8,41 +7,41 @@ namespace ChallengeMk2.DataBase
 {
     public class SQLiteDataService : ILocalDataService
     {
-        SQLiteConnection database;
+        SQLiteAsyncConnection database;
 
-        public void Initialize()
+        public async Task Initialize()
         {
             if (database == null)
             {
-                database = new SQLiteConnection(DatabaseParameters.DbPath, DatabaseParameters.DbFlags);
-                database.CreateTable<StarSystemDbItem>();
+                database = new SQLiteAsyncConnection(DatabaseParameters.DbPath, DatabaseParameters.DbFlags);
+                await database.CreateTableAsync<StarSystemDbItem>();
             }
         }
 
-        public List<StarSystemDbItem> GetFullDb()
+        public async Task<List<StarSystemDbItem>> GetFullDb()
         {
-            return database.Table<StarSystemDbItem>().ToList();
+            return await database.Table<StarSystemDbItem>().ToListAsync();

```

Comment:
What happens if I have forgotten to call `Initialize` before calling another method ? How to improve that experience ?

Reply:
> What happens if I have forgotten to call Initialize before calling another method ?

Reply:
yep, an exception will be throw because there is no db or table already :(

Reply:
i can check if db is null with a null-conditional :
return await database?.Table<StarSystemDbItem>().ToListAsync();

Reply:
i can do that for each method from SQLiteDataService